### DISCUSSION TITLE
Community fluxes

### DISCUSSION THREAD

**Discussion by beva005** [micom-dev/micom#106] [Ideas]
Hello,

I have a suggestion for the better understanding of this great tool. 

I think it would be more helpful for the user if the reactions id within com.reactions would correspond with the ids given to the computed fluxes after sol=com.optimize. 

For example, if we only have a community of 2 species:
If we have the glucose exchange reaction in our models (EX_GLC_D_e), the community model would have transport reactions from the extracellular compartment to the medium compartment for each species (glc_d__e_speciesi <-> glc_d_m):  EX_GLC_D_e__species1; EX_GLC_D_e__species2 and the one defined in the medium: EX_GLC_m. However, sol.fluxes would give three columns corresponding to species 1, species 2 and medium, but the id of those reactions would be EX_GLC_D_e. While I understand that the list of reactions would be huge when there are many species and having different columns helps to reduce this, I would suggest changing this EX_GLC_D_e id to something that makes clear that it is a transport reaction. Otherwise, it might be confusing at first.

Thanks,
Sara

**Comment by cdiener** [comment]
Hey, thanks for the suggestion. I don't get it completely because it seems what you are suggesting is what is already taking place. Like for the example:

```
In [9]: sol.fluxes.EX_ac_e
Out[9]: 
compartment
Escherichia_coli_1    0.224978
Escherichia_coli_2    0.140000
Escherichia_coli_3    0.149131
Escherichia_coli_4    0.194445
medium                     NaN
Name: EX_ac_e, dtype: float64
```

It does not list it for the medium because `EX_ac_m` means something different, because it is the transport into the external environment, not the transport into a particular taxon. However, this format is marked as deprecated and will [likely be replaced by a new representation](https://github.com/micom-dev/micom/issues/34) that improves the distinction between internal and transport fluxes and also incorporates annotations. Hopefully, that new format includes the information you are missing in the current version.
